### PR TITLE
Fix Codex mock provider routing

### LINKED ISSUE #28: MockProviderHost: codex-mock hangs because /v1/responses route is not implemented
[Gautam's Dev bot]

## Symptom

In `samples/LmStreaming.Sample`, selecting the `codex-mock` provider and sending a user message causes the chat UI to hang on the "Stop" state. The WebSocket never receives a `done` event; only the user turn is persisted.

Reproduction: PR #26 preview env, choose `codex-mock`, type "hello", send.

## Root cause

Codex CLI 0.128 (and current versions) defaults to the OpenAI **Responses API** (`POST /v1/responses`). `samples/MockProviderHost/MockProviderHostBuilder.cs:95-99` only routes:

```csharp
app.MapPost("/v1/chat/completions", …);
app.MapPost("/v1/messages", …);
```

Codex's app-server hits the unrouted `/v1/responses` path; the request 404s, the JSON-RPC turn never resolves, and the agent loop hangs. See `src/CodexSdkProvider/Transport/CodexAppServerTransport.cs:84-106` for how `OPENAI_BASE_URL` is plumbed (no `wire_api` override today).

## Already acknowledged in tree

`samples/MockProviderHost/README.md:175-178`:

> Newer Copilot model paths (GPT-5 family) use `/v1/responses`. If the probe finds the CLI defaults to that route on the version under test, the follow-up E2E issue must add a `/v1/responses` route to `MockProviderHostBuilder` before E2E coverage is possible.

Codex hits the same gap as Copilot's GPT-5 path.

## Proposed fixes (pick one)

**(a) Add `/v1/responses` route to `MockProviderHostBuilder`** that adapts `ScriptedSseResponder` output to OpenAI Responses-API SSE shape. Larger surface (new wire format in `src/LmTestUtils/TestMode/`), but future-proofs for the Copilot GPT-5 path called out in the README.

**(b) Override Codex wire format to chat-completions when a mock base URL is set** by passing `-c model_providers.openai.wire_api="chat"` to `codex app-server` in `CodexAppServerTransport.cs:87` when `baseUrl` is non-null. ~10 LoC, targeted, but couples the Codex transport to mock-mode behavior and skips Responses-API-only Codex features.

## Definition of done

- `codex-mock` provider produces a streaming reply end-to-end against the in-process mock host on the LmStreaming.Sample preview env.
- A regression test (E2E or unit) covers the chosen wire path so the failure mode can't silently return.

## Linked PR

#26 — where the limitation was reported and a UX banner was added pointing here.

### PR DESCRIPTION
**Pull request by gautam-achieveai**
## Summary
Fixes Codex(Mock) routing so a mock `BaseUrl` is passed to Codex app-server as an explicit OpenAI Responses-compatible custom model provider instead of relying only on `OPENAI_BASE_URL`.

This allows Codex app-server to use `MockProviderHost` for Responses/WebSocket traffic and removes the stale `codex-mock` known-limitation banner from the sample provider registry.

## Changes
- Add explicit `modelProvider` and `config.model_provider/model_providers` wiring when `CodexBridgeInitOptions.BaseUrl` is set.
- Configure the custom provider with `wire_api = responses`, `requires_openai_auth = false`, and `supports_websockets = true`.
- Add regression tests for start/resume app-server config generation.
- Remove the outdated Codex(Mock) limitation message from `LmStreaming.Sample`.

## Testing
- `dotnet format whitespace LmDotnetTools.sln --verify-no-changes`
- `dotnet test tests\CodexSdkProvider.Tests\CodexSdkProvider.Tests.csproj --no-restore`
- `dotnet test tests\LmStreaming.Sample.Tests\LmStreaming.Sample.Tests.csproj --no-restore --filter "FullyQualifiedName~ProviderRegistryTests|FullyQualifiedName~CodexAgentLoopTests"`
- `dotnet test tests\MockProviderHost.Tests\MockProviderHost.Tests.csproj --no-restore --filter "FullyQualifiedName~Responses"`
- `dotnet build LmDotnetTools.sln --no-restore`
- Live smoke: `CodexSdkClient` + `MockProviderHost` consumed the scripted mock turn (`remaining_turns=0`) and emitted visible mock text.

## Related Issues
Fixes #28

### CODE REVIEW

**Review by gautam-achieveai — COMMENTED** [review]
**[Gautam's review bot]** — Deep review of PR #36

## Context
Fixes #28 — Codex app-server hangs when routed through MockProviderHost because the mock host doesn't implement `/v1/responses`. This PR configures an explicit custom model provider in the Codex app-server config when `BaseUrl` is set, with `wire_api = "responses"`, `supports_websockets = true`, and `requires_openai_auth = false`.

## Review Summary

**Approach is correct and well-executed.** The fix adds two clean helper methods (`ApplyModelProviderOverride` and `AddResponsesModelProviderConfig`) that inject explicit model provider config only when `BaseUrl` is present. Both thread-start and thread-resume paths are handled consistently. The limitation banner cleanup in `ProviderRegistry` is the right follow-through.

### What's good
- Clean early-return guard pattern in both helpers
- `env_key` logic correctly conditional on `ApiKey` presence
- Both start/resume paths tested via `[Theory]`
- Separate test for the no-API-key edge case

### Findings

| # | Severity | Finding |
|---|----------|---------|
| 1 | MEDIUM | Missing negative test — no assertion that `modelProvider`/`model_providers` are absent when `BaseUrl` is null. Protects against accidental config leakage into non-mock scenarios. |
| 2 | LOW | Reflection-based `InvokeThreadParams` helper is fragile — signature changes break at runtime, not compile time. Acceptable tradeoff for now. |
| 3 | INFO | `OPENAI_BASE_URL` env var (set in `CodexAppServerTransport`) is now redundant when the explicit model provider config is present. Not a bug, just an observation. |

### Tests
- `CodexAppServerRequestConfigTests`: 3/3 pass
- `ProviderRegistryTests.KnownLimitation_*`: 1/1 pass

**Verdict: APPROVE** — The fix is clean, well-scoped, and correctly resolves the hang. The missing negative test is a suggestion, not a blocker.

File: tests/CodexSdkProvider.Tests/Agents/CodexAppServerRequestConfigTests.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            methodName,
+            BindingFlags.Instance | BindingFlags.NonPublic)
+            ?? throw new MissingMethodException(nameof(CodexSdkClient), methodName);
+
+        return method.Invoke(client, [options])
+                .Should()
+                .BeAssignableTo<Dictionary<string, object?>>()
+                .Subject;
+    }
+}
```
Comment (gautam-achieveai) [inline tests/CodexSdkProvider.Tests/Agents/CodexAppServerRequestConfigTests.cs:93]: **[Gautam's review bot]** MEDIUM — Consider adding a negative test that verifies when `BaseUrl` is null/empty, neither `modelProvider` is set on the params dictionary nor `model_provider`/`model_providers` appear in the config. This guards against regression where the custom provider config accidentally leaks into production (non-mock) scenarios.
